android/src/main/java/com/pspdfkit/views/PdfView.java

```diff
@@ -239,8 +248,18 @@ public void setMenuItemGroupingRule(@NonNull MenuItemGroupingRule groupingRule)
     private void setupFragment() {
         if (fragmentTag != null && configuration != null && document != null) {
             PdfFragment pdfFragment = (PdfFragment) fragmentManager.findFragmentByTag(fragmentTag);
+            if (pdfFragment != null && pdfFragment.getArguments().getInt(ARG_ROOT_ID) != internalId) {
```

Comment:
Better add also a safety check against `null` for the bundle (i.e. `pdfFragment.getArguments() != null`)